compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -1145,7 +1145,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         // are the same function and their parameters have a LUB.
                         match self.commit_if_ok(|_| {
                             self.at(cause, self.param_env).lub(
-                                DefineOpaqueTypes::No,
+                                DefineOpaqueTypes::Yes,
```

Comment:
`DefineOpaqueTypes::Yes` inside of a `commit_if_ok` likely means that this can cause type inference breakage.

When coercing function definitions we may have previously used the `lub` further down as this `lub` here failed. This would have then caused us to return a function pointer (i think), whereas we now return an `FnDef`.

I would like to see a crater run here to allow us to more eagerly detect any regressions from it